x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/color_stops_utils.js

```diff
@@ -14,37 +28,114 @@ export function removeRow(colorStops, index) {
   return [...colorStops.slice(0, index), ...colorStops.slice(index + 1)];
 }
 
-export function addRow(colorStops, index) {
+export function addOrdinalRow(colorStops, index) {
+  return addRow(colorStops, index, COLOR_MAP_TYPE.ORDINAL);
+}
+
+export function addCategoricalRow(colorStops, index) {
+  return addRow(colorStops, index, COLOR_MAP_TYPE.CATEGORICAL);
+}
+
+export function addRow(colorStops, index, colorMapType) {
   const currentStop = colorStops[index].stop;
-  let delta = 1;
-  if (index === colorStops.length - 1) {
-    // Adding row to end of list.
-    if (index !== 0) {
-      const prevStop = colorStops[index - 1].stop;
-      delta = currentStop - prevStop;
+
+  let nextValue;
+  if (colorMapType === COLOR_MAP_TYPE.ORDINAL) {
+    let delta = 1;
+    if (index === colorStops.length - 1) {
+      // Adding row to end of list.
+      if (index !== 0) {
+        const prevStop = colorStops[index - 1].stop;
+        delta = currentStop - prevStop;
+      }
+    } else {
+      // Adding row in middle of list.
+      const nextStop = colorStops[index + 1].stop;
+      delta = (nextStop - currentStop) / 2;
     }
+    nextValue = currentStop + delta;
   } else {
-    // Adding row in middle of list.
-    const nextStop = colorStops[index + 1].stop;
-    delta = (nextStop - currentStop) / 2;
+    nextValue = currentStop === '' ? currentStop + 'a' : '';
   }
-
   const newRow = {
-    stop: currentStop + delta,
-    color: '#FF0000',
+    stop: nextValue,
+    color: DEFAULT_CUSTOM_COLOR,
   };
   return [...colorStops.slice(0, index + 1), newRow, ...colorStops.slice(index + 1)];
 }
 
+export function getDeleteButton(onRemove) {
+  return (
+    <EuiButtonIcon
+      iconType="trash"
+      color="danger"
+      aria-label={i18n.translate('xpack.maps.styles.colorStops.deleteButtonAriaLabel', {
+        defaultMessage: 'Delete',
+      })}
+      title={i18n.translate('xpack.maps.styles.colorStops.deleteButtonLabel', {
+        defaultMessage: 'Delete',
+      })}
+      onClick={onRemove}
+    />
+  );
+}
+
+export function getColorInput(colorStops, onColorChange, color) {
```

Comment:
Same for `getColorInput`and `getColorStopRow`. Just move these directly into `ColorStops` component